### ISSUE TITLE
Tipo do produto e classificação fiscal não são setadas corretamente.

### ISSUE THREAD

**Issue by danimaribeiro** [odoo-brazil/l10n-brazil#119]
Por exemplo ao criar uma fatura diretamente, tanto de fornecedor quanto de cliente, quando não se tem uma regra fiscal já configurada por padrão, fazendo com que a posição fiscal não carregue, ocorre um erro com o onchange do produto, a classificação fiscal e o tipo do produto não carregam corretamente.
No exemplo abaixo é possível ver que o serviço manteve o tipo do produto "Produto",e também que a posição fiscal está vazia, mesmo após escolher a posição fiscal a classificação não atualiza.

![erro tipo de produto e classificacao fiscal](https://cloud.githubusercontent.com/assets/854781/10324308/101ed53a-6c5f-11e5-9760-ccf844ee938e.png)

Já na situação abaixo, existe uma regra fiscal cadastrada fazendo com que a posição fiscal seja setada automaticamente, com isto a classificação e o tipo são setados corretamente.

![onchange correto](https://cloud.githubusercontent.com/assets/854781/10324409/b8288be0-6c5f-11e5-917f-cf3b9cf37d3d.png)

Quando a posição fiscal não carrega corretamente, a execução cai dentro deste if:
https://github.com/odoo-brazil/l10n-brazil/blob/8.0/l10n_br_account_product/account_invoice.py#L758

Fazendo com que o seguinte código que está no mesmo método não seja executado:
https://github.com/odoo-brazil/l10n-brazil/blob/8.0/l10n_br_account_product/account_invoice.py#L813

Uma saída seria remover o código que seta a classificação fiscal de dentro do metodo _validate_taxes e colocar ele no onchange do produto de onde é chamado:
https://github.com/odoo-brazil/l10n-brazil/blob/8.0/l10n_br_account/account_invoice.py#L552

@mileo @mstuttgart @cmsalmeida @mackilem

**Comment by mstuttgart** [comment]
Acredito que seja uma boa abordagem. O código que atribui valor para a "classificação fiscal" não depende do valor armazenado na "posição fiscal". Então não há motivos para ele ficar sem valor caso a "posição fiscal" não esteja configurada.

**Comment by Mackilem** [comment]
Concordo com o Michell, a classificação fiscal é sempre necessária. Não
consigo imaginar uma solução melhor, acho que é isso mesmo Danimar

Em ter, 6 de out de 2015 às 20:37, Michell Stuttgart <
notifications@github.com> escreveu:

**Comment by pauloneda** [comment]
@danimaribeiro 
Aproveitando o tópico.
Se os impostos estiverem setado na empresa, ao adicionar um item a Fatura as CSTS são carregadas corretamente mas ao salvar a Fatura elas desaparecem, porém se os impostos estiverem no NCM é carregado e salvo corretamente as CSTS.
